src/components/app/app.jsx

```diff
@@ -4,58 +4,60 @@ import BurgerConstructor from '../burger-constructor/burger-constructor'
-const ingredientContext = createContext(null)
+import OrderDetails from '../modal-content/order-details/order-details'
+import IngredientDetails from '../modal-content/ingredient-details/ingredient-details'
+import ModalOverlay from '../modal-overlay/modal-overlay'
+
+import { getFromApi } from '../../utils/api'
+const IngredientContext = createContext(null)
+const ConstructorContext = createContext(null)
 
 export default function App() {
   const [ingredient, setIngredient] = useState(false)
   const [isOrderDetailsOpened, setIsOrderDetailsOpened] = useState(false)
   const [ingredientsData, setIngredientsData] = useState(false)
+  const [constructorItems, setConstructorItems] = useState([])
 
   useEffect(() => {
-    fetch(dataUrl)
-      .then((res) => res.json())
-      .then((result) => result.data)
-      .then((data) => setIngredientsData(data))
-      .catch((err) => console.log(err))
+    getFromApi('ingredients').then((data) => setIngredientsData(data))
   }, [])
 
   const closeAllModals = () => {
     setIsOrderDetailsOpened(false)
     setIngredient(false)
   }
-  const handleEscKeydown = (e) => e.key === 'Escape' && closeAllModals()
-
   return (
     <>
       <AppHeader />
+      <ConstructorContext.Provider value={{ constructorItems, setConstructorItems }}>
+        {ingredientsData && (
+          <div className={style.content}>
+            <IngredientContext.Provider value={{ ingredientsData, ingredient, setIngredient }}>
+              <BurgerIngredients />
+              <BurgerConstructor onSubmit={setIsOrderDetailsOpened} />
+            </IngredientContext.Provider>
+          </div>
+        )}
 
-      {ingredientsData && (
-        <div className={style.content}>
-          <ingredientContext.Provider value={{ ingredientsData, ingredient, setIngredient }}>
-            <BurgerIngredients />
-            <BurgerConstructor onSubmit={setIsOrderDetailsOpened} />
-          </ingredientContext.Provider>
-        </div>
-      )}
-
-      {isOrderDetailsOpened && (
-        <Modal onOverlayClick={closeAllModals} onEscDown={handleEscKeydown}>
-          <OrderDetails onSubmit={closeAllModals} />
-        </Modal>
-      )}
+        {isOrderDetailsOpened && (
+          <>
+            <Modal onEscKeyDown={closeAllModals}>
+              <OrderDetails onSubmit={closeAllModals} />
+            </Modal>
+            <ModalOverlay onClick={closeAllModals} />
+          </>
+        )}
+      </ConstructorContext.Provider>
 
       {ingredient && (
-        <Modal
-          title="Детали ингредиента"
-          onOverlayClick={closeAllModals}
-          onEscDown={handleEscKeydown}
-        >
-          <IngredientDetails item={ingredient} />
-        </Modal>
+        <>
+          <Modal title="Детали ингредиента" onEscKeyDown={closeAllModals}>
+            <IngredientDetails item={ingredient} />
+          </Modal>
+          <ModalOverlay onClick={closeAllModals} />
+        </>
```

Comment:
Можно немного порефакторить модалки: перенести компонент ModalOverlay внутрь компонента Modal и убрать из ModalOverlay портал - это позволит везде не писать несколько раз компонент ModalOverlay
И рендер модального окна станет выглядеть вот так:
```suggestion
          <Modal title="Детали ингредиента" onEscKeyDown={closeAllModals}>
            <IngredientDetails item={ingredient} />
          </Modal>
```